packages/mui-material/src/ButtonBase/ButtonBase.js

```diff
@@ -47,24 +47,28 @@ export const ButtonBaseRoot = styled('button', {
   borderRadius: 0,
   padding: 0, // Remove the padding in Firefox
   cursor: 'pointer',
-  userSelect: 'none',
   verticalAlign: 'middle',
   MozAppearance: 'none', // Reset
   WebkitAppearance: 'none', // Reset
   textDecoration: 'none',
   // So we take precedent over the style of a native <a /> element.
   color: 'inherit',
+  '&:active': {
+    userSelect: 'none',
+  },
   '&::-moz-focus-inner': {
     borderStyle: 'none', // Remove Firefox dotted outline.
   },
   [`&.${buttonBaseClasses.disabled}`]: {
-    pointerEvents: 'none', // Disable link interactions
     cursor: 'default',
+    ...(!ownerState.isButton && {
+      pointerEvents: 'none', // Disable link interactions
```

Comment:
If this line is support to disable link interactions, wouldn't it be better if we have a check about whether the component rendered is a link instead of checking that it is not button? For e.g. having <div role="button" /> will fall into this condition.

Reply:
It doesn't apply only to links, it applies to all other elements as well, I updated the comment to be more accurate.